core/services/workflows/engine.go

```diff
@@ -1167,10 +1167,14 @@ func (e *Engine) Close() error {
 		// any triggers to ensure no new executions are triggered,
 		// then we'll close down any background goroutines,
 		// and finally, we'll deregister any workflow steps.
+
+		// deregistering a trigger is done via don2don.
+		// the trigger may be already deregistered by the time this
+		// engine instance is Closed by the syncer.
```

Comment:
How so?
In the remote case, triggerSubscriber->UnregisterTrigger() never returns an error...

I think the change is valid - I don't think we should stop the Close() flow if deregistering fails. But the comment is not accurate.